File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Model/Logs/TimeoutLog.java
package com.siliconlabs.bledemo.Browser.Model.Logs;

import android.bluetooth.BluetoothDevice;

import com.siliconlabs.bledemo.Browser.Model.LogType;

public class TimeoutLog extends Log {

    public TimeoutLog(BluetoothDevice device) {
        setLogTime(getTime());
        setLogInfo(device.getAddress() + "(" + getDeviceName(device.getName()) + "): "
                + "Connection timeout");
        setLogType(LogType.INFO); //malo wazne
        setDeviceAddress(device.getAddress());
    }

    public TimeoutLog() {
        setLogTime(getTime());
        setLogInfo("Connection timeout");
        setLogType(LogType.INFO); //malo wazne
    }

}

File: mobile/src/main/java/com/siliconlabs/bledemo/Browser/Models/Logs/TimeoutLog.kt
package com.siliconlabs.bledemo.Browser.Models.Logs

import android.bluetooth.BluetoothDevice
import com.siliconlabs.bledemo.Browser.Models.LogType

class TimeoutLog : Log {
    constructor(device: BluetoothDevice) {
        logTime = getTime()
        logInfo = (device.address + " (" + getDeviceName(device.name) + "): "
                + "Connection timeout")
        logType = LogType.INFO
        deviceAddress = device.address
    }

    constructor() {
        logTime = getTime()
        logInfo = "Connection timeout"
        logType = LogType.INFO
    }
}